javagen/src/main/java/com/azure/autorest/template/ServiceSyncClientTemplate.java

```diff
@@ -129,25 +130,28 @@ protected void writeClass(AsyncSyncClient syncClient, JavaClass classBlock, Java
   protected void writeMethods(AsyncSyncClient syncClient, JavaClass classBlock) {
     final ServiceClient serviceClient = syncClient.getServiceClient();
     final MethodGroupClient methodGroupClient = syncClient.getMethodGroupClient();
-    final boolean wrapServiceClient = methodGroupClient == null;
-    if (wrapServiceClient) {
-      serviceClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
-    } else {
-      methodGroupClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
+
+    List<ClientMethod> clientMethods = serviceClient.getClientMethods();
+    if (CoreUtils.isNullOrEmpty(clientMethods)) {
+      clientMethods = methodGroupClient.getClientMethods();
     }
 
+    clientMethods.stream()
+        .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
+        .filter(clientMethod -> !clientMethod.isImplementationOnly())
+        .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
+        // service client methods generate overloads that are with and without context that aren't needed
+        // for wrapper public clients
+        .filter(clientMethod -> (JavaSettings.getInstance().isDataPlaneClient())
+                || ((clientMethod.getName().contains("WithResponse") && clientMethod.getParametersDeclaration().contains("Context context"))
+                || ((!clientMethod.getName().contains("WithResponse")
```

Comment:
DPG has method "WIthResponse" and "RequestOptions".

Many visibility/hidden is now moved to `ClientMethodMapper.methodVisibility`. You might try adjust there. https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java#L1047-L1052

Particularly, since I was not sure about the visibility of the sync-stack mode, I keep it same as what I saw before the refactor (moving the control of visibility to this place). You can set it to `NOT_VISIBLE` I think it will be private in client impl, and not be shown in wrapper clients.